### PR TITLE
Requests that match a non-cached variant should go to the origin.

### LINKED ISSUE #740: Default values for Language (and others?)
https://github.com/httpwg/http-extensions/compare/master...jyasskin:incomplete-cache-example shows an example of an origin server that contains English, French, and German versions of a resource, a cache that has cached the English and French versions, a request that asks for German or Spanish, and, I believe, permission in the spec for the cache to return the English version instead of asking the origin server for the German version.

This happens because https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html#content-language appends the first Variant to the list of requested languages even if the list of requested languages overlaps with the available variants. Is this a mistake in the algorithm, or intentional for some reason?

### PR DESCRIPTION
**Pull request by jyasskin**
I've also added examples of the new behavior.

This is an attempt to fix #740.

### CODE REVIEW

File: draft-ietf-httpbis-variants.md [hunk 1]
```diff
@@ -525,7 +597,7 @@ To perform content negotiation for Accept-Encoding given a request-value and ava
 
 1. Let preferred-available be an empty list.
 2. Let preferred-codings be a list of the codings in the request-value, ordered by their weight, highest to lowest, as per Section 5.3.1 of {{!RFC7231}} (omitting any coding with a weight of 0). If "Accept-Encoding" is not present or empty, preferred-codings will be empty. If a coding lacks an explicit weight, an implementation MAY assign one.
-3. If "identity" is not a member of preferred-codings, append "identity".
+3. If "identity" is not a member of preferred-codings, and it is not specifically excluded by the request-value stating either "identity;q=0" or "*;q=0" without a more specific entry for "identity", append "identity" to preferred-codings.
```
Comment (mnot) [inline draft-ietf-httpbis-variants.md]: Encoding is not different; the server can ignore the client's preferences here too.

File: draft-ietf-httpbis-variants.md [hunk 2]
```diff
@@ -339,6 +339,76 @@ Variant-Key: fr, gzip
...
+~~~
+
+If the cache contains responses with the following Variant-Keys:
+
+~~~ example
+Variant-Key: fr
+Variant-Key: en
+~~~
+
+Then the cache needs to forward the request to the origin server, since it offered an exact match to the client's request, `Variant-Key: de`.
```
[the remark this answers is not in the document]
Comment (jyasskin) [reply draft-ietf-httpbis-variants.md]: "it" was the origin server, but I'll update to your wording tomorrow morning.

File: draft-ietf-httpbis-variants.md (hunk 1 above)
Comment (jyasskin) [reply draft-ietf-httpbis-variants.md]: That is, I should undo this change? That's fine with me.

Comment (mnot) [reply draft-ietf-httpbis-variants.md]: Yes.

File: draft-ietf-httpbis-variants.md [hunk 3]
```diff
@@ -339,6 +339,76 @@ Variant-Key: fr, gzip
...
+
+~~~ example
+[
+  ["de"]         // prefers German; will not accept English
+]
+~~~
+
+{{cache}}{:format="title"} returns:
+
+~~~ example
```
[the remark this answers is not in the document]
Comment (jyasskin) [reply draft-ietf-httpbis-variants.md]: Nope, I can remove this code block.

File: draft-ietf-httpbis-variants.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jyasskin) [reply draft-ietf-httpbis-variants.md]: Done.

**Pull request merged**